integration/hibernate-6.0/src/main/java/com/blazebit/persistence/integration/hibernate/Hibernate60Access.java

```diff
@@ -214,4 +232,33 @@ public ParameterTranslations createParameterTranslations(List<ParameterSpecifica
         return new ParameterTranslationsImpl(queryParameterSpecifications);
     }
 
+    /**
+     * Iterator over {@code ScrollableResultsImplementor}.
+     *
+     * @author Jan-Willem Gmelig Meyling
+     * @since 1.6.2
+     */
+    private static class ScrollableResultsIterator implements Iterator<Object> {
+        private final ScrollableResultsImplementor scrollableResults;
+
+        public ScrollableResultsIterator(ScrollableResultsImplementor scrollableResults) {
+            this.scrollableResults = scrollableResults;
+        }
+
+        @Override
+        public boolean hasNext() {
+            return !scrollableResults.isClosed() && scrollableResults.next();
```

Comment:
AFAIU `next()` will advance the result, so I think it is more appropriate to check `!isLast()` here and in all other implementations

Reply:
On further thought, it might be better to eagerly call `next()` in the constructor and check `scrollableResults.get() != null`

Reply:
I have some questions of my own about the desiredness of a check to `isClosed` here. I think invoking `next` on a (likely unexpectedly) closed scrollable result should throw an exception for sure. However, I just copied the implementation from https://github.com/hibernate/hibernate-orm/blob/main/hibernate-core/src/main/java/org/hibernate/query/internal/ScrollableResultsIterator.java#L19-L47 as I assumed there were good reasons to do this (in fact, the entire `ScrollableResultsImplementor` interface appears to be introduced merely for this purpose...)

See: https://github.com/hibernate/hibernate-orm/commit/7cae5ba95bf355d1023c25820e45eaf5f984bcb7
And: https://hibernate.atlassian.net/browse/HHH-9340